src/models/thread.ts

```diff
@@ -556,14 +556,18 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
 
     /**
      * Determines if the given user has read a particular event ID with the known
-     * history of the room. This is not a definitive check as it relies only on
+     * history of the thread. This is not a definitive check as it relies only on
      * what is available to the thread at the time of execution.
      * @param userId - The user ID to check the read state of.
      * @param eventId - The event ID to check if the user read.
      * @returns True if the user has read the event, false otherwise.
      */
     public hasUserReadEvent(userId: string, eventId: string): boolean {
         if (userId === this.client.getUserId()) {
+            // Consider an event read if it's part of a thread that has no
+            // read receipts and has no notifications. It is likely that it is
```

Comment:
This doesn't seem to match the logic?

> part of a thread that has no read receipts

We don't seem to be looking at read receipts on the thread itself?

> has no notifications

... or considering notifications?